source/renderer/app/stores/StakingStore.js

```diff
@@ -202,6 +202,8 @@ export default class StakingStore extends Store {
         // Retrieves the API update
         this.smashServerLoading = true;
         await this.updateSmashSettingsRequest.execute(smashServerUrl);
+        // Resets the Stake Pools list request
+        this.stakePoolsRequest.reset();
```

Comment:
@daniloprates I believe it could happen that there is an already running request.
In that case you should await it and then reset it. Perhaps something like this could do the trick:

```
if (this.stakePoolsRequest.isExecuting) {
  await this.stakePoolsRequest;
}
this.stakePoolsRequest.reset();
```

Reply:
@nikolaglumac this solution would actually keep the old information being displayed :/ 

I believe we won't need any further change, but if we do, I think we'll need them to actually cancel or discard the API result from old requests at the API level.